sonar-plugin-api/src/main/java/org/sonar/api/ServerComponent.java

```diff
@@ -19,15 +19,17 @@
  */
 package org.sonar.api;
 
+import org.sonar.api.server.ComputeEngineSide;
 import org.sonar.api.server.ServerSide;
 
 /**
  * Same than {@link org.sonar.api.BatchComponent} but for server-side components.
  *
  * @since 2.2
- * @deprecated since 5.2 use {@link ServerSide} annotation
+ * @deprecated in 5.2. Use {@link ServerSide} or {@link ComputeEngineSide} annotation instead.
  */
 @Deprecated
 @ServerSide
+@ComputeEngineSide
```

Comment:
this was a conservative move.

but we call also assume that no deprecated class/interface/annotation should be use to code Components for the CE

I'll remove it